configx/provider.go

```diff
@@ -333,6 +337,9 @@ func (p *Provider) Set(key string, value interface{}) error {
 }
 
 func (p *Provider) BoolF(key string, fallback bool) bool {
+	p.l.RLock()
```

Comment:
Are these really needed? I was under the impression that koanf reading is lock free. Or is the problem the setters?

Reply:
Exactly, when we replace the koanf instance we get a race condition. https://app.circleci.com/pipelines/github/ory/keto/1308/workflows/238f2513-8e2e-4182-b911-5fdf94c19658/jobs/8684?invite=true#step-106-311